templates/components/multiagent/typescript/workflow/factory.ts

```diff
@@ -18,8 +23,52 @@ class WriteEvent extends WorkflowEvent<{
   isGood: boolean;
 }> {}
 class ReviewEvent extends WorkflowEvent<{ input: string }> {}
+class PublishEvent extends WorkflowEvent<{ input: string }> {}
+
+const prepareChatHistory = (chatHistory: ChatMessage[]) => {
```

Comment:
@thucpn @leehuwuj how about moving this to templates/components/llamaindex ? then @thucpn's code can extend and reuse it

Reply:
yes, I thought about adding it to the llamaindex component too, but since it was just my PR using it, I decided to keep it here

Reply:
@leehuwuj can you then move this in #328